olp-cpp-sdk-core/include/olp/core/context/Context.h

```diff
@@ -33,57 +33,108 @@
 namespace olp {
 namespace context {
 /**
- * @brief The Context class represents the application context
+ * @brief Represents the application context.
  *
- * Client application must initialize its Context before using any other
- * functionality of the library.
+ * Before your application uses any other functionality of the library,
+ * initialize the `Context` class.
  */
-
 struct ContextData;
 
+/**
+ * @brief Used only for the Android environment to correctly initialize
+ * the `NetworkAndroid` class.
+ *
+ * Initialize the `Context` class before you send network requests in
+ * the Android environment.
+ * To use the `Context` class, cerate the `Scope` object first.
+ */
 class CORE_API Context {
  public:
+  /**
+   * @brief Called when the `Context` object is initialized.
+   *
+   * Controlled by the `Scope` class.
+   *
+   * @see `Scope` for more information.
+   */
   using InitializedCallback = std::function<void()>;
+  /**
+   * @brief Called when the `Context` object is deinitialized.
+   *
+   * Controlled by the `Scope` class.
+   *
+   * @see `Scope` for more information.
+   */
   using DeinitializedCallback = std::function<void()>;
 
   /**
-   * @brief addInitializeCallbacks Register functions to be called after the
-   * context is initialized and destroyed.
-   * @param initCallback
-   * @param deinitCalback
+   * @brief Registers functions that are called when the context is initialized
+   * and destroyed.
+   *
+   * @param initCallback The `InitializedCallback` instance.
+   * @param deinitCalback The `DeinitializedCallback` instance.
    */
   static void addInitializeCallbacks(InitializedCallback initCallback,
                                      DeinitializedCallback deinitCalback);
 
   /**
-   * @brief The Scope class initializes the context in its constructor (if not
-   * already initialized) and deinitializes in the destructor (when no other
-   * Scope instances exist). Instead of calling Context::init() and
-   * Context::deinit() manually, simply instantiate a Context::Scope() object.
+   * @brief Initializes the `Context` class in its constructor (if it is not
+   * already initialized) and deinitializes in its destructor (if there are
+   * no other `Scope` instances).
+   *
+   * Instead of calling `Context::init()` and `Context::deinit()` manually,
+   * instantiate the `Context::Scope()` object.
    */
   class CORE_API Scope {
    public:
-    /// see Context::init()
+    /**
+     * @brief Creates the `Scope` instance.
+     *
+     * Used to initialize the `Context` class.
```

Comment:
`<rant>` I see such sentences starting with `Used to ...` all across the SDK and while they might make sense in the library context. They sound weird, and I'm not even sure that such a sentence is grammatically correct.`</rant>`
What do you think should fix it some other day?

Reply:
In the descriptions, it is a common practice not to repeat the class/method/function name and not to use phrases like 'this method", "this function", and so on. This sentence is a shorter version of  "This constructor is used to initialize the `Context` class". 

For more information, please see the [API Reference Code Comments](https://developers.google.com/style/api-reference-comments) section in the 
Developer Documentation Style Guide.

Reply:
However, we can put it in the following way: "The `Scope` instance is used to initialize the `Context` class."

Reply:
I was just asking a question, but thanks for changing that quick and pro-actively anyway :)